src/Core/Math/Sphere.js

```diff
@@ -0,0 +1,54 @@
+import THREE from 'THREE';
+import defaultValue from 'Core/defaultValue';
+
+
+function Sphere(center,radius) {
+
+	this.center = defaultValue(center,new THREE.Vector3());
+	this.radius = defaultValue(radius,1.0);
+
+}
+
+Sphere.prototype.constructor = Sphere;
+
+Sphere.prototype.setCenter = function(center) {
+
+	this.center.copy(center);
+};
+
+Sphere.prototype.setRadius = function(radius) {
+
+	this.radius = radius;
+};
+
+var vector = new THREE.Vector3();
+
+Sphere.prototype.intersectWithRay = function(ray) {
+
+    var pc = ray.closestPointToPoint(this.center);
+    var a = pc.length(),d,b;
```

Comment:
We can now use `let` instead of `var` to get proper variables scopes (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let)